sklearn/impute.py

```diff
@@ -531,12 +554,42 @@ def _get_missing_features_info(self, X):
                 X = X.toarray()
             imputer_mask = _get_mask(X, self.missing_values)
             features_with_missing = np.flatnonzero(imputer_mask.sum(axis=0))
+            imputer_mask, features_with_missing = self._drop_constant_columns(
+                imputer_mask, features_with_missing
+            )
 
             if self.sparse is True:
                 imputer_mask = sparse.csc_matrix(imputer_mask)
 
         return imputer_mask, features_with_missing
 
+    @staticmethod
+    def _drop_constant_columns(imputer_mask, features_with_missing):
+        """
+        In case we have a column with totally missing  values, drop it,
+        because constant column is not informative.
+        :param imputer_mask:
+        :param features_with_missing:
+        :return:
+        """
+        is_sparce = sparse.issparse(imputer_mask)
+
+        if is_sparce:
+            imputer_mask = imputer_mask.toarray()
+
```

Comment:
This conversion is not a good idea and might lead to memory error on large matrices.

A better way would be to convert it in `csc` format and then keep the non full missing columns. Here is the way to do it:
```python
if self.sparse:
    sparse_format = imputer_mask.format
    imputer_mask = csc_matrix(imputer_mask)

non_fully_missing_indexes = np.where(
    features_with_missing != imputer_mask.shape[0])[0]
imputer_mask = imputer_mask[:, non_fully_missing_indexes]
features_with_missing = features_with_missing[non_fully_missing_indexes]

if self.sparse:
    imputer_mask = csr_matrix(imputer_mask) if sparse_format is 'csr' else imputer_mask
```